src/unit_tests/syscheckd/test_fim_sync.c

```diff
@@ -119,6 +159,213 @@ static int teardown_json_payload(void **state) {
     return 0;
 }
 
+static int setup_str_pair(void **state) {
+    str_pair_t *new;
+    os_calloc(1, sizeof(str_pair_t), new);
+    os_strdup("first", new->first);
+    os_strdup("last", new->last);
+
+    *state = new;
+    return 0;
+}
+
+
+static int teardown_str_pair(void **state) {
+    str_pair_t *new = *state;
+
+    os_free(new->first);
+    os_free(new->last);
+    os_free(new);
+
+    return 0;
+}
+
+static int setup_fim_entry(void **state) {
+    pair_entry_str_t *data = NULL;
+
+
+    data = malloc(sizeof(pair_entry_str_t));
+    data->entry = malloc(sizeof(fim_entry));
+    data->entry->file_entry.data = malloc(sizeof(fim_file_data));
+
+    data->entry->type = FIM_TYPE_FILE;
+    data->entry->file_entry.path = strdup("start");
+
+    data->entry->file_entry.data->size = 1501;
+    data->entry->file_entry.data->perm = strdup("0666");
+    data->entry->file_entry.data->attributes = strdup("rw-rw-rw-");
+    data->entry->file_entry.data->uid = strdup("101");
+    data->entry->file_entry.data->gid = strdup("1001");
+    data->entry->file_entry.data->user_name = strdup("test1");
+    data->entry->file_entry.data->group_name = strdup("testing1");
+    data->entry->file_entry.data->mtime = 1570184224;
+    data->entry->file_entry.data->inode = 606061;
+    strcpy(data->entry->file_entry.data->hash_md5, "3691689a513ace7e508297b583d7550d");
+    strcpy(data->entry->file_entry.data->hash_sha1, "07f05add1049244e7e75ad0f54f24d8094cd8f8b");
+    strcpy(data->entry->file_entry.data->hash_sha256, "672a8ceaea40a441f0268ca9bbb33e9959643c6262667b61fbe57694df224d40");
+    data->entry->file_entry.data->mode = FIM_REALTIME;
+    data->entry->file_entry.data->last_event = 1570184221;
+    data->entry->file_entry.data->dev = 12345678;
+    data->entry->file_entry.data->scanned = 123456;
+    data->entry->file_entry.data->options = 511;
+    strcpy(data->entry->file_entry.data->checksum, "07f05add1049244e7e71ad0f54f24d8094cd8f8b");
+
+    cJSON *json  = fim_entry_json("start", data->entry);
+    data->str = cJSON_PrintUnformatted(json);
+
+    free(json);
+
+    *state = data;
+
+    return data == NULL;
+}
+
+static int teardown_fim_entry(void **state) {
+    pair_entry_str_t *data = *state;
+    // data->str and data->entry should be freed
+    free(data);
+    return 0;
+}
+
+/* Auxiliar functions */
+
+static void expect_fim_db_get_first_row_error(const fdb_t *db, int type, char *path) {
+    expect_function_call(__wrap_pthread_mutex_lock);
+
+    expect_value(__wrap_fim_db_get_first_path, fim_sql, db);
+    expect_value(__wrap_fim_db_get_first_path, type, type);
+    will_return(__wrap_fim_db_get_first_path, path);
+    will_return(__wrap_fim_db_get_first_path, FIMDB_ERR);
+
+    expect_string(__wrap__merror, formatted_msg, "(6706): Couldn't get FIRST FILE row's path.");
```

Comment:
Use the `FIM_DB_ERROR_GET_ROW_PATH` constant instead of the hardcoded message. It will be easier to maintain in the future in case the message changes. To add `FIRST FILE` in the message, you can use `snprintf`.